File: Java/everywhere.java

// https://open.kattis.com/problems/everywhere
import java.util.HashSet;
import java.util.Scanner;

public class everywhere {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int cases = sc.nextInt();
        while (cases-- > 0) {
            HashSet<String> cities = new HashSet<>();
            int trips = sc.nextInt();
            while (trips-- > 0)
                cities.add(sc.next());
            System.out.println(cities.size());
        }
        sc.close();
    }
}

File: C#/everywhere.cs
using System;
using System.Linq;

public class everywhere {
    public static void Main (string[] args) {
        int cases = int.Parse (Console.ReadLine ());
        while (cases-- > 0) {
            int num = int.Parse (Console.ReadLine ());
            string[] cities = new string[num];
            for (int x = 0; x < num; x++)
                cities[x] = Console.ReadLine ();
            Console.WriteLine (cities.Distinct ().Count ());
        }
    }
}